pkg/sql/sem/builtins/builtins.go

```diff
@@ -3188,3 +3205,49 @@ func truncateTimestamp(
 	toTime := time.Date(year, month, day, hour, min, sec, nsec, loc)
 	return tree.MakeDTimestampTZ(toTime, time.Microsecond), nil
 }
+
+func asJSON(d tree.Datum) (json.JSON, error) {
+	switch t := d.(type) {
+	case *tree.DBool:
+		return json.FromBool(bool(*t)), nil
+	case *tree.DInt:
+		return json.FromInt(int(*t)), nil
+	case *tree.DFloat:
+		return json.FromFloat64(float64(*t))
+	case *tree.DDecimal:
+		return json.FromDecimal(t.Decimal), nil
+	case *tree.DString:
+		return json.FromString(string(*t)), nil
+	case *tree.DCollatedString:
+		return json.FromString(t.Contents), nil
+	case *tree.DJSON:
+		return t.JSON, nil
+	case *tree.DArray:
+		jsons := make([]json.JSON, t.Len())
+		for i, e := range t.Array {
+			var err error
+			jsons[i], err = asJSON(e)
+			if err != nil {
+				return nil, err
+			}
+		}
+		return json.FromArrayOfJSON(jsons), nil
+	case *tree.DTuple:
+		kvPairs := make([]json.KeyValuePair, t.D.Len())
+		for i, e := range t.D {
+			j, err := asJSON(e)
+			if err != nil {
+				return nil, err
+			}
+			kvPairs[i] = json.MakeKeyValuePair(fmt.Sprintf("f%d", i+1), j)
```

Comment:
I think rather than exposing this extra stuff, I'd prefer to just construct a `map` and then use `MakeJSON`, especially since this is a pretty niche use-case, and I think having fewer exposed methods of creating JSON values is preferable.